src/Utils/MX/Methods/LoadRandomMap.as

```diff
@@ -35,6 +35,96 @@ namespace MX
             return;
         }
 
+        if (!PluginSettings::UseLengthChecksInRequests) {
+            if ((RMC::IsRunning || RMC::IsStarting) && !PluginSettings::CustomRules) {
+                if (!(
+                    map.LengthName == "15 secs" || map.LengthName == "30 secs" || map.LengthName == "45 secs" || map.LengthName == "1 min" || map.LengthName == "1 m 15 s" || 
+                    map.LengthName == "1 m 30 s" || map.LengthName == "1 m 45 s" || map.LengthName == "2 min" || map.LengthName == "2 m 30 s" || map.LengthName == "3 min"
+                )) {
+                    Log::Warn("Map is too long, retrying...");
+                    PreloadRandomMap();
+                    return;
+                }
+            } else if (PluginSettings::MapLength != "Anything") {
+                int requiredLength = PluginSettings::SearchingMapLengthsMilliseconds[PluginSettings::SearchingMapLengths.Find(PluginSettings::MapLength)];
+                switch (PluginSettings::SearchingMapLengthOperators.Find(PluginSettings::MapLengthOperator)) {
+                    case 0:  // exact is prorbably a bit too strict so we'll allow a 5 second difference
```

Comment:
5 seconds is too short, can you try making 30 second difference ?

Reply:
30 seconds seems a bit extreme considering the lowest configurable interval is 15 seconds, so for me it seems more logical to have it range within 15 seconds? With 30 seconds difference you can a 45 second map when you set only 15 second maps

Reply:
15 seconds is ok 👍